src/thinkfan.cpp

```diff
@@ -299,6 +303,20 @@ void TemperatureState::add_temp(int t)
 	int diff = t - *temp_;
 	*temp_ = t;
 
+	if (diff == 0) {
+		if (*noise_counter < NOISE_MEASUREMENT) {
```

Comment:
Could you explain why do we need the whole vector `noise_counters` when we only use the first element?
If I understand your logic, then you just need to keep one static count, increase that count every time `diff==0` and compare that count with a fix threshold. In this case, you don't even need `noise_counter` and `noise_counters` as member variables.